jans-linux-setup/jans_setup/setup_app/utils/base.py

```
@@ -381,20 +382,48 @@ def mylog(logs):
-        except:
-             mylog("Error downloading {}. Download will be re-tried once more".format(url))
-             download_tries += 1
-             time.sleep(1)
+            download_ok = True
+        except (HTTPError, URLError, OSError):
+            retry_sec = random.randint(1,4)
+            mylog(f"Error downloading {url}. Download will be re-tried once more in {retry_sec} seconds.")
+            download_tries += 1
+            if download_tries >=3:
+                time.sleep(retry_sec)
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)